src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -1336,6 +1336,7 @@ public DelegatedLdapUserDetails loadUserByUsername(String username) throws Usern
                 // Add those, as done in LdapAuthenticationProvider.createUserDetails().
                     LdapUserDetailsImpl.Essence user = new LdapUserDetailsImpl.Essence(ldapUser);
                     user.setUsername(username);
+                    user.setDn(ldapUser.getNameInNamespace()); // otherwise the DN is missing the DC
```

Comment:
Is this right?

Reply:
No clue, but if it works it works right? :/